src/transformers/models/character_bert/modeling_character_bert.py

```diff
@@ -0,0 +1,1999 @@
+    size (batch_size, sequence_length, 50) and returns (batch_size, sequence_length, hidden_size), where hidden_size is
+    typically 768.
+    """
+
+    def __init__(self, config):
+        super().__init__()
+        self.character_embeddings_dim = config.character_embeddings_dim
+        self.cnn_activation = config.cnn_activation
+        self.cnn_filters = config.cnn_filters
+        self.num_highway_layers = config.num_highway_layers
+        self.max_word_length = config.max_word_length
+        self.hidden_size = config.hidden_size
+        # NOTE: this is the 256 possible utf-8 bytes + special slots for the
+        # [CLS]/[SEP]/[PAD]/[MASK] characters as well as beginning/end of
+        # word symbols and character padding for short words -> total of 263
+        self.character_vocab_size = 263
+        self._init_weights()
+
+    def get_output_dim(self):
+        return self.hidden_size
+
+    def _init_weights(self):
+        self._init_char_embedding()
+        self._init_cnn_weights()
+        self._init_highway()
+        self._init_projection()
+
+    def _init_char_embedding(self):
+        weights = torch.empty((self.character_vocab_size, self.character_embeddings_dim))
+        nn.init.normal_(weights)
+        weights[0].fill_(0.0)  # the all zeros token is used for padding token sequences
+        weights[PAD_CHARACTER_ID + 1].fill_(0.0)  # this character id is used for padding character sequences
+        self._char_embedding_weights = torch.nn.Parameter(torch.FloatTensor(weights), requires_grad=True)
+
+    def _init_cnn_weights(self):
+        convolutions = []
+        for i, (width, num) in enumerate(self.cnn_filters):
+            conv = torch.nn.Conv1d(
+                in_channels=self.character_embeddings_dim, out_channels=num, kernel_size=width, bias=True
+            )
+            conv.weight.requires_grad = True
+            conv.bias.requires_grad = True
+            convolutions.append(conv)
+            self.add_module(f"char_conv_{i}", conv)
+        self._convolutions = convolutions
+
+    def _init_highway(self):
+        # the highway layers have same dimensionality as the number of cnn filters
+        n_filters = sum(f[1] for f in self.cnn_filters)
+        self._highways = Highway(n_filters, self.num_highway_layers, activation=nn.functional.relu)
+        for k in range(self.num_highway_layers):
+            # The AllenNLP highway is one matrix multplication with concatenation of
+            # transform and carry weights.
+            self._highways._layers[k].weight.requires_grad = True
+            self._highways._layers[k].bias.requires_grad = True
+
+    def _init_projection(self):
+        n_filters = sum(f[1] for f in self.cnn_filters)
+        self._projection = torch.nn.Linear(n_filters, self.hidden_size, bias=True)
+        self._projection.weight.requires_grad = True
+        self._projection.bias.requires_grad = True
```

Comment:
wheights should be initialized in ther `PreTrained` class (See llama code or mamba code for a reference)